src/Components/ErrorBoundary/ErrorBoundary.jsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { Text, Center } from '@chakra-ui/react';
+
+class ErrorBoundary extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { hasError: false };
+  }
+
+  static getDerivedStateFromError() {
+    // Update state so the next render will show the fallback UI.
+    return { hasError: true };
+  }
+
+  componentDidCatch(error, errorInfo) {
+    // You can also log the error to an error reporting service
+    // logErrorToMyService(error, errorInfo);
+    this.setState({
+      error: error,
```

Comment:
those state properties don't exist in a constructor, and you check for hasErros in the render method which will be always false since it have not updated to true.

P.S.: check that example by Den Abramov https://codepen.io/gaearon/pen/wqvxGa?editors=0010